firewood/src/storage/linear/mod.rs

```diff
@@ -71,44 +68,15 @@ mod proposed;
 #[cfg(test)]
 mod tests;
 
-#[derive(Debug)]
-pub(super) struct LinearStore<S: ReadLinearStore> {
-    state: S,
-}
-
 /// All linearstores support reads
-pub(super) trait ReadLinearStore: Send + Sync + Debug {
+pub(super) trait ReadLinearStore: Debug {
     fn stream_from(&self, addr: u64) -> Result<Box<dyn Read + '_>, Error>;
     fn size(&self) -> Result<u64, Error>;
 }
 
-impl ReadLinearStore for Arc<dyn ReadLinearStore> {
-    fn stream_from(&self, addr: u64) -> Result<Box<dyn Read + '_>, Error> {
-        self.deref().stream_from(addr)
-    }
-
-    fn size(&self) -> Result<u64, Error> {
-        self.deref().size()
-    }
-}
-
 /// Some linear stores support updates
 pub(super) trait WriteLinearStore: Debug {
     fn write(&mut self, offset: u64, object: &[u8]) -> Result<usize, Error>;
 }
 
-impl<ReadWrite: ReadLinearStore + Debug> WriteLinearStore for LinearStore<ReadWrite> {
-    fn write(&mut self, _offset: u64, _bytes: &[u8]) -> Result<usize, Error> {
-        todo!()
-    }
-}
-
-impl<S: ReadLinearStore> ReadLinearStore for LinearStore<S> {
-    fn stream_from(&self, addr: u64) -> Result<Box<dyn Read + '_>, Error> {
-        self.state.stream_from(addr)
-    }
-
-    fn size(&self) -> Result<u64, Error> {
-        todo!()
-    }
-}
+pub(super) trait ReadWriteLinearStore: ReadLinearStore + WriteLinearStore + Debug {}
```

Comment:
As Richard pointed out, you can't have a `WriteLinearStore` that doesn't allow for reads (completely meaningless). So, lets just do:
```rust
pub(super) trait WriteLinearStore: ReadLinearStore {
    fn write(&mut self, offset: u64, object: &[u8]) -> Result<usize, Error>;
}
```
There is no need to repeat `Debug` or `Send` or `Sync` since trait bounds are inherited.